core/citrus-base/src/main/java/org/citrusframework/functions/core/AdvancedRandomNumberFunction.java

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.citrusframework.functions.core;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.List;
+import java.util.concurrent.ThreadLocalRandom;
+
+import org.citrusframework.context.TestContext;
+import org.citrusframework.exceptions.CitrusRuntimeException;
+import org.citrusframework.exceptions.InvalidFunctionUsageException;
+import org.citrusframework.functions.Function;
+
+import static java.lang.String.format;
+
+/**
+ * A function for generating random double values with specified decimal places and range. This
+ * function includes options to specify the number of decimal places, minimum and maximum values,
+ * and whether to include or exclude the minimum and maximum values.
+ * <p>
+ * Parameters:
+ * <ol>
+ *     <li>Decimal places: The number of decimal places in the generated random number (optional, default: 0). Note that definition of 0 results in an integer.</li>
+ *     <li>Min value: The minimum value for the generated random number (optional, default: Double.MIN_VALUE).</li>
+ *     <li>Max value: The maximum value for the generated random number (optional, default: Double.MAX_VALUE).</li>
```

Comment:
these are not the actual `DEFAULT_MAX_VALUE` and `DEFAULT_MIN_VALUE` being used (see https://github.com/citrusframework/citrus/pull/1318/files#diff-167bf384ca89ef57c71db3fb84f936a1a26b31d840611638192d2ce6e97f525cR54), or am I being misleaded? if so, the variable naming could be improved, I think. I am also wondering why these `DEFAULT_*` variables must be `public`?

Reply:
I set these default values to avoid generation of incredibly large numbers by default. Still it is possible to set min/max to higher values. Internally BigDecimal is used as value. The naming in the javadoc is wrong. I adjusted it. Also made the constants private.